tests/test_modeling_tf_gptj.py

```diff
@@ -0,0 +1,382 @@
+            "input_ids": input_ids,
+            "token_type_ids": token_type_ids,
+            "attention_mask": input_mask,
+        }
+        return config, inputs_dict
+
+
+@require_tf
+class TFGPTJModelTest(TFModelTesterMixin, TFCoreModelTesterMixin, unittest.TestCase):
+
+    all_model_classes = (
+        (TFGPTJForCausalLM, TFGPTJForSequenceClassification, TFGPTJForQuestionAnswering, TFGPTJModel)
+        if is_tf_available()
+        else ()
+    )
+
+    all_generative_model_classes = (TFGPTJForCausalLM,) if is_tf_available() else ()
+    test_onnx = False
+    test_pruning = False
+    test_missing_keys = False
+    test_head_masking = False
+
+    def setUp(self):
+        self.model_tester = TFGPTJModelTester(self)
+        self.config_tester = ConfigTester(self, config_class=GPTJConfig, n_embd=37)
+
+    def test_config(self):
+        self.config_tester.run_common_tests()
+
+    def test_gptj_model(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_gptj_model(*config_and_inputs)
+
+    def test_gptj_model_past(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_gptj_model_past(*config_and_inputs)
+
+    def test_gptj_model_att_mask_past(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_gptj_model_attention_mask_past(*config_and_inputs)
+
+    def test_gptj_model_past_large_inputs(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_gptj_model_past_large_inputs(*config_and_inputs)
+
+    def test_gptj_lm_head_model(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_gptj_lm_head_model(*config_and_inputs)
+
+    def test_model_common_attributes(self):
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+
+        for model_class in self.all_model_classes:
+            model = model_class(config)
+            assert isinstance(model.get_input_embeddings(), tf.keras.layers.Layer)
+
+            if model_class in self.all_generative_model_classes:
+                x = model.get_output_embeddings()
+                assert isinstance(x, tf.keras.layers.Layer)
+                name = model.get_bias()
+                assert name is None
+            else:
+                x = model.get_output_embeddings()
+                assert x is None
+                name = model.get_bias()
+                assert name is None
+
+    @slow
+    def test_model_from_pretrained(self):
+        model = TFGPTJModel.from_pretrained("EleutherAI/gpt-j-6B", from_pt=True)
+        self.assertIsNotNone(model)
+
+
+@require_tf
+class TFGPTJModelLanguageGenerationTest(unittest.TestCase):
+    @tooslow
+    def test_lm_generate_gptj(self):
```

Comment:
This test has the `@tooslow` decorator, meaning it won't be run in our CI. @stancld can you confirm that it pass the test?

Also, I'd like to ask you to add some tests that are in the PT suite, for completeness: `test_gptj_sample`, `test_gptj_sample_max_time` (this one might fail, tag it with `@tooslow` if it does) and `test_batch_generation` (which should be a `TFGPTJModelLanguageGenerationTest` test) :)

Reply:
@gante 
I can confirm `test_lm_generate_gptj` passes successfully. Currently, I've been debugging `test_batch_generation` where `tokne_type_ids` has not change the output even though it should according to PT test.

Reply:
> where tokne_type_ids has not change the output even though it should according to PT test.

This may be due to our implementation of TF `generate()`, which is not as sophisticated as our PT counterpart. I'm going to confirm whether it is the case.

Reply:
I can confirm -- it is a problem on our side. Some context: we are currently refactoring TF `generate()` so it gets the same structure as in PT, then we will add missing functionality. Using `token_type_ids` is one of the missing functionalities.

Meanwhile, I'd like to ask you to copy the `prepare_inputs_for_generation()` from [TF's GPT2](https://github.com/huggingface/transformers/blob/master/src/transformers/models/gpt2/modeling_tf_gpt2.py#L854) into TF GPTJ's `prepare_inputs_for_generation()` -- including the TODO, so we can track it. As for the test, you can set the expected outcome to the outcome you're getting locally 👍